send/ad_mu

```diff
@@ -1025,22 +984,32 @@ saveStoredUserStates;
 		@studentsSecondPart{@studentsIndexes[$upperBound+1..$#studentsIndexes]} = @{$students}{@studentsIndexes[$upperBound+1..$#studentsIndexes]};
 	}
 
-	add_to_license_group($ad_entries_group_map{$employeeDN}, $ad_state->{$employeeDN}, $employees);
-	add_to_license_group($ad_entries_group_map{$alumniDN}, $ad_state->{$alumniDN}, $alumni);
-	add_to_license_group($ad_entries_group_map{$studentDN}, $ad_state->{$studentDN}, \%studentsFirstPart);
-	add_to_license_group($ad_entries_group_map{$student2DN}, $ad_state->{$student2DN}, \%studentsSecondPart);
+	my %add_result;
+	my %remove_result;
 
+	$add_result{$employeeDN} = add_to_license_group($ad_entries_group_map{$employeeDN}, $ad_state->{$employeeDN}, $employees);
+	$add_result{$alumniDN} = add_to_license_group($ad_entries_group_map{$alumniDN}, $ad_state->{$alumniDN}, $alumni);
+	$add_result{$studentDN} = add_to_license_group($ad_entries_group_map{$studentDN}, $ad_state->{$studentDN}, \%studentsFirstPart);
+	$add_result{$student2DN} = add_to_license_group($ad_entries_group_map{$student2DN}, $ad_state->{$student2DN}, \%studentsSecondPart);
 
-	remove_from_license_group($ad_entries_group_map{$employeeDN}, $ad_state->{$employeeDN}, $employees);
-	remove_from_license_group($ad_entries_group_map{$alumniDN}, $ad_state->{$alumniDN}, $alumni);
-	remove_from_license_group($ad_entries_group_map{$studentDN}, $ad_state->{$studentDN}, \%studentsFirstPart);
-	remove_from_license_group($ad_entries_group_map{$student2DN}, $ad_state->{$student2DN}, \%studentsSecondPart);
+
+	$remove_result{$employeeDN} = remove_from_license_group($ad_entries_group_map{$employeeDN}, $ad_state->{$employeeDN}, $employees);
+	$remove_result{$alumniDN} = remove_from_license_group($ad_entries_group_map{$alumniDN}, $ad_state->{$alumniDN}, $alumni);
+	$remove_result{$studentDN} = remove_from_license_group($ad_entries_group_map{$studentDN}, $ad_state->{$studentDN}, \%studentsFirstPart);
+	$remove_result{$student2DN} = remove_from_license_group($ad_entries_group_map{$student2DN}, $ad_state->{$student2DN}, \%studentsSecondPart);
+
+	foreach ($employeeDN, $alumniDN, $studentDN, $student2DN) {
+		if ($add_result{$_} eq "errors" or $remove_result{$_} eq "errors") {
+			$counter_group_updated_with_errors++;
+		} elsif ($add_result{$_} eq "changed" or $remove_result{$_} eq "changed") {
+			$counter_group_updated++;
+		}
+	}
 
 	# process specific license groups
 	foreach my $group_dn (sort keys %{$perun_state}) {
 		unless (($group_dn eq $employeeDN) or ($group_dn eq $studentDN) or ($group_dn eq $alumniDN) or ($group_dn eq $student2DN)) {
-			add_to_license_group($ad_entries_group_map{$group_dn}, $ad_state->{$group_dn}, $perun_state->{$group_dn});
-			remove_from_license_group($ad_entries_group_map{$group_dn}, $ad_state->{$group_dn}, $perun_state->{$group_dn});
+			update_group_membership($ad_entries_group_map{$group_dn}, $ad_state->{$group_dn}, $perun_state->{$group_dn});
```

Comment:
Former implementation of ``add_to_license_group`` and ``remove_from_license_group`` exits the script on LDAP change failure. By using new implementation, it continues the script.

It might not be desired, since failure in adding of member and success of removing will cause inconsistency for the mentioned "process inside LDAP" and it might result in storing this state in local caches and sending unnecessary notifications to the user.

Also ``update_group_membership`` here uses generic methods to add/remove members, so it probably won't be subject of possible fixes for managing license groups.

Reply:
I've added check after processing the main license groups, which ends the script if one of them fails.